File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.cpp
class Trie {
    struct Node{
        int end = false;
        array<Node*,26> nxt{}; // don't forget the curly brackets!
        Node() = default;
    };

    Node* root;

public:
    Trie() {
        root = new Node();
    }
    
    void insert(string word) {
        Node* cur = root;
        for (char ch : word){
            int i = ch-'a';
            if (!cur->nxt[i])
                cur->nxt[i] = new Node();
            cur = cur->nxt[i];
        }
        cur->end = true;
    }
    
    bool search(string word) {
        Node* cur = root;
        for (char ch : word){
            int i = ch-'a';
            if (!cur->nxt[i])
                return false;
            cur = cur->nxt[i];
        }
        return cur->end;
    }
    
    bool startsWith(string prefix) {
        Node* cur = root;
        for (char ch : prefix){
            int i = ch-'a';
            if (!cur->nxt[i])
                return false;
            cur = cur->nxt[i];
        }
        return true;
    }
};

/**
 * Your Trie object will be instantiated and called as such:
 * Trie* obj = new Trie();
 * obj->insert(word);
 * bool param_2 = obj->search(word);
 * bool param_3 = obj->startsWith(prefix);
 */

File: 0208-implement-trie-prefix-tree/0208-implement-trie-prefix-tree.py
class Node:
    def __init__(self):
        self.end = False
        self.nxt = [None] * 26

class Trie:

    def __init__(self):
        self.root = Node()

    def insert(self, word: str) -> None:
        cur = self.root
        for ch in word:
            i = ord(ch)-ord('a')
            if not cur.nxt[i]:
                cur.nxt[i] = Node()
            cur = cur.nxt[i]
        cur.end = True

    def search(self, word: str) -> bool:
        cur = self.root
        for ch in word:
            i = ord(ch)-ord('a')
            if not cur.nxt[i]:
                return False
            cur = cur.nxt[i]
        return cur.end

    def startsWith(self, prefix: str) -> bool:
        cur = self.root
        for ch in prefix:
            i = ord(ch)-ord('a')
            if not cur.nxt[i]:
                return False
            cur = cur.nxt[i]
        return True


# Your Trie object will be instantiated and called as such:
# obj = Trie()
# obj.insert(word)
# param_2 = obj.search(word)
# param_3 = obj.startsWith(prefix)